### DISCUSSION TITLE
How to hide visible area emitters while keeping the illumination?

### DISCUSSION THREAD

**Discussion by Galaxeaaa** [mitsuba-renderer/mitsuba3#1136] [Q&A]
Hi! I've been working on a lighting optimization project. I tried to represent the near-field lighting as a grid of cube emitters, and optimize the intensity of each cube emitter to match the ground truth.

However, there will be a visibility jump near zero intensity. So I'm thinking of a workaround that, simply not rendering the emitter itself. I found a parameter `hide_emitters` for some built-in integrators, but from the source code and my experiments, this parameter only hide environment parameters.

I wonder if there's a built-in solution for my need. If not, how would you recommend to implement this? Will it be better to implement an Emitter or an Integrator?

Thank you!

**Comment by njroussel** [comment]
Hi @Galaxeaaa 

Indeed, there should always be a `hide_emitters` option in the integrators, which should do exactly that. However, as you noticed it only applies to environment emitters. It's unclear what you'd want in other use cases: for an `area` emitter do you completely ignore the shape or do you just make it non-emissive?

Anyway, I definitely think that the easiest way to do this is by modifiying the integrator you're using. The mask should be something along the lines of `si.is_valid() & si.shape.is_emitter() & dr.eq(depth, 0)`
